frontend/src/lib/components/Annotations/AnnotationMarker.tsx

```diff
@@ -51,6 +52,19 @@ export function AnnotationMarker({
     const [applyAll, setApplyAll] = useState(true)
     const [textAreaVisible, setTextAreaVisible] = useState(false)
     const [hovered, setHovered] = useState(false)
+    const { reportAnnotationViewed } = useActions(eventUsageLogic)
+
+    const visible = focused || (!dynamic && hovered)
+
+    useEffect(() => {
+        if (visible) {
+            reportAnnotationViewed(annotations)
+        } else {
+            reportAnnotationViewed(null)
+            /* We report a null value to cancel (if applicable) the report because the annotation was closed */
+        }
+    }, [visible])
```

Comment:
I'm not sure this works - usually the dependency array should consist of useState or props, but I see no reason why it wouldn't work.

Reply:
Seems to work okay 😉